sycl/include/sycl/detail/vector_arith.hpp

```diff
@@ -355,38 +398,14 @@ template <typename DataT, int NumElements> class vec_arith_common {
   template <typename T = DataT>
   friend std::enable_if_t<!detail::is_vgenfloat_v<T>, vec_t>
   operator~(const vec_t &Rhs) {
-#ifdef __SYCL_DEVICE_ONLY__
-    auto extVec = sycl::bit_cast<typename vec_t::vector_t>(Rhs);
-    vec_t Ret{~extVec};
-    if constexpr (std::is_same_v<DataT, bool>) {
-      ConvertToDataT(Ret);
-    }
-    return Ret;
-#else
-    vec_t Ret{};
-    for (size_t I = 0; I < NumElements; ++I) {
-      Ret[I] = ~Rhs[I];
-    }
-    return Ret;
-#endif
-  }
-
-#ifdef __SYCL_DEVICE_ONLY__
-  using vec_bool_t = vec<bool, NumElements>;
-  // Required only for std::bool.
-  static void ConvertToDataT(vec_bool_t &Ret) {
-    for (size_t I = 0; I < NumElements; ++I) {
-      Ret[I] = bit_cast<int8_t>(Ret[I]) != 0;
-    }
+    return VecOperators::apply_unary_op<std::bit_not<void>>(Rhs);
```

Comment:
omg, if true, I love this.  Stupid `operator~`  was as huge pain. we need to make sure it's fully tested, lin/win , bool, std::byte (and not applicable to float IIRC).

Reply:
I'm running local CTS (bin/test_vector* tests) in addition to pre-commit CI.